.gitlab/source_test/linux.yml

```diff
@@ -17,56 +17,87 @@
 
 tests_deb-x64-py3:
-  extends: .linux_tests_preparation
-  stage: source_test
+  extends: .linux_tests
+  needs: ["linux_x64_go_deps"]
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: ["runner:main", "size:2xlarge"]
   variables:
     PYTHON_RUNTIMES: '3'
     CONDA_ENV: ddpy3
-  script:
-    - inv -e test --race --profile --python-runtimes "$PYTHON_RUNTIMES" --cpus 4
 
 tests_rpm-x64-py2:
-  extends: .linux_tests_preparation
-  stage: source_test
+  extends: .linux_tests
+  needs: ["linux_x64_go_deps"]
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/rpm_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: ["runner:main", "size:2xlarge"]
   variables:
     PYTHON_RUNTIMES: '2'
     CONDA_ENV: ddpy2
-  script:
     # Exclude systemd because it cannot succeed on Centos 6: the image doesn't have the shared object required by
     # https://github.com/coreos/go-systemd/blob/c8cc474ba8655dfbdb0ac7fcc09b7faf5b643caf/sdjournal/functions.go#L46
     # This is OK because the test on systemd still runs on the debian image above
-    - inv -e test --race --profile --python-runtimes "$PYTHON_RUNTIMES" --cpus 4 --build-exclude=systemd
+    EXTRA_OPTS: '--build-exclude=systemd'
 
 tests_rpm-x64-py3:
-  extends: .linux_tests_preparation
-  stage: source_test
+  extends: .linux_tests
+  needs: ["linux_x64_go_deps"]
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/rpm_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: ["runner:main", "size:2xlarge"]
   variables:
     PYTHON_RUNTIMES: '3'
     CONDA_ENV: ddpy3
-  script:
     # Exclude systemd because it cannot succeed on Centos 6: the image doesn't have the shared object required by
     # https://github.com/coreos/go-systemd/blob/c8cc474ba8655dfbdb0ac7fcc09b7faf5b643caf/sdjournal/functions.go#L46
     # This is OK because the test on systemd still runs on the debian image above
-    - inv -e test --race --profile --python-runtimes "$PYTHON_RUNTIMES" --cpus 4 --build-exclude=systemd
+    EXTRA_OPTS: '--build-exclude=systemd'
+
+tests_deb-arm64-py2:
+  extends: .linux_tests
+  needs: ["linux_arm64_go_deps"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_arm64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:docker-arm", "platform:arm64"]
+  variables:
+    PYTHON_RUNTIMES: '2'
+    CONDA_ENV: ddpy2
+
+tests_deb-arm64-py3:
+  extends: .linux_tests
+  needs: ["linux_arm64_go_deps"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_arm64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:docker-arm", "platform:arm64"]
+  variables:
+    PYTHON_RUNTIMES: '3'
+    CONDA_ENV: ddpy3
+
+tests_rpm-arm64-py2:
+  extends: .linux_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/rpm_arm64:$DATADOG_AGENT_BUILDIMAGES
```

Comment:
Did you forget to specify `needs` here as you do in the other tests?

Reply:
Yes I did, should be fixed now :)